File: test_cases/python/claude/05.py
def run_tests(solution_module):
    f = solution_module.longestValidParentheses
    # all open — no valid pairs
    assert f('(((') == 0
    # nested valid
    assert f('(())') == 4
    # multiple separated valid segments, longest is 4
    assert f('()(()') == 2
    # entire string is valid
    assert f('()()') == 4
    # single close bracket
    assert f(')') == 0

File: test_cases/c/claude/05.c
#include <assert.h>
#include <stdio.h>

int longestValidParentheses(char* s);

int main(void) {
    assert(longestValidParentheses("(()") == 2);
    assert(longestValidParentheses(")()())") == 4);
    assert(longestValidParentheses("") == 0);
    assert(longestValidParentheses("()(()") == 2);
    assert(longestValidParentheses("()(())") == 6);
    return 0;
}